### DISCUSSION TITLE
Having issues with layout shifts in nextjs application

### DISCUSSION THREAD

**Discussion by simon-lehmann** [mantinedev/mantine#3766] [Q&A]
Hello, I have created a webpage (nextjs/mantine) but having issues with big layout shifts on rerender without cache, all of my pages are static. 
I went through the documentation and updated my custom document:
![image](https://user-images.githubusercontent.com/57042505/224764580-979ca8a2-9cbc-411c-806e-28baa18c5e01.png)

When putting a breakpoint in a page i can visualize the layout shift in an image:
![image](https://user-images.githubusercontent.com/57042505/224765966-71304d18-cbfd-4015-8354-a7a3002d5085.png)

It seems like the hydration is happening before the styles are loaded.
This issue is also noticeable in the lighthouse score in the initial page load.

Any ideas of what I'm missing from that little information?

**Comment by rtivital** [comment]
You can use one of the next.js templates as a reference, they use the latest versions of mantine and next.js and do not have ssr issues.

**Reply by simon-lehmann** [reply]
Thanks, the template gave me direction on where to debug. The root cause was a non-computable css property. A background with a copied linear gradient directly copied from Figma
`background:
      'linear-gradient(360deg, #040c17 0%, rgba(4, 12, 23, 0) 27%), linear-gradient(180deg, #040c17 30.29%, rgba(4, 12, 23, 0) 71.62%)',`

It works on client, but somehow breaks when preprocessing with emotion on the server

**Reply by simon-lehmann** [reply]
Unfortunately, I have to take it back, it still does not work for me. I will try to set up a repo which reproduces this issue
